librz/include/rz_util/rz_path.h

```diff
@@ -2,23 +2,39 @@
 #define RZ_UTIL_PATH_H_
 
 #include <rz_types.h>
+#include <rz_th.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-RZ_API void rz_path_set_prefix(RZ_NONNULL const char *path);
-RZ_API RZ_OWN char *rz_path_prefix(RZ_NULLABLE const char *path);
-RZ_API RZ_OWN char *rz_path_incdir(void);
-RZ_API RZ_OWN char *rz_path_bindir(void);
-RZ_API RZ_OWN char *rz_path_libdir(void);
-
-RZ_API RZ_OWN char *rz_path_system(RZ_NULLABLE const char *path);
+/**
+ * \brief Holds Rizin's path prefix and synchronization utilities.
+ *
+ * This structure stores the computed prefix of Rizin's path, along with a flag
+ * indicating whether the prefix has already been computed to avoid redundant work.
+ * It also includes a mutex to ensure thread-safe access when computing or accessing the prefix.
+ */
+typedef struct rz_path_t {
+	char *prefix; ///< the computed path prefix.
+	bool prefix_searched; ///< flag indicating if the prefix has already been computed.
+	RzThreadLock *prefix_mutex; ///< mutex to protect access to the prefix in multithreaded contexts.
+} RzPath;
+
+#ifdef RZ_API
+
+RZ_API void rz_path_set_prefix(RZ_BORROW RZ_NONNULL RzPath *sys_path, RZ_NONNULL const char *path);
+RZ_API RZ_OWN char *rz_path_prefix(RZ_BORROW RZ_NONNULL RzPath *sys_path);
+RZ_API RZ_OWN char *rz_path_incdir(RZ_BORROW RZ_NULLABLE RzPath *sys_path);
+RZ_API RZ_OWN char *rz_path_bindir(RZ_BORROW RZ_NULLABLE RzPath *sys_path);
+RZ_API RZ_OWN char *rz_path_libdir(RZ_BORROW RZ_NULLABLE RzPath *sys_path);
+
+RZ_API RZ_OWN char *rz_path_system(RZ_BORROW RZ_NULLABLE RzPath *sys_path, RZ_NULLABLE const char *path);
 RZ_API RZ_OWN char *rz_path_extra(RZ_NULLABLE const char *path);
 RZ_API RZ_OWN char *rz_path_home_prefix(RZ_NULLABLE const char *path);
 RZ_API RZ_OWN char *rz_path_home(RZ_NULLABLE const char *path);
 
-RZ_API RZ_OWN char *rz_path_system_rc(void);
+RZ_API RZ_OWN char *rz_path_system_rc(RZ_BORROW RZ_NULLABLE RzPath *sys_path);
```

Comment:
Please make all remove all the borrows (do not remove the one from `set_prefix`) and add `const` before `RzPath *`.
Also why some are nullable and some not nullable?